tests/commonlib/kubernetes.py

```diff
@@ -179,7 +187,83 @@ def delete_resources(self, resource_type: str, **kwargs):
     def patch_resources(self, resource_type: str, **kwargs):
         """
         """
-        return self.dispatch_patch[resource_type](**kwargs)
+        if resource_type == RESOURCE_POD:
+            patch_body = kwargs.pop('body')
+
+            pod = self.get_resource(resource_type, **kwargs)
+            self.delete_resources(resource_type=resource_type, **kwargs)
+            deleted = self.wait_for_resource(resource_type=resource_type, status_list=['DELETED'], **kwargs)
+
+            if not deleted:
+                raise ValueError(f'could not delete Pod: {kwargs}')
+
+            pod = self.create_patched_resource(resource_type, patch_body)
+
+            return pod
+
+        else:
+            return self.dispatch_patch[resource_type](**kwargs)
+    
+    def create_patched_resource(self, patch_resource_type, patch_body):
+        file_path = Path(__file__).parent / '../deploy/mock-pod.yml'
+        k8s_resources = get_k8s_yaml_objects(file_path=file_path)
+
+        patch_metadata = patch_body['metadata']
+        patch_relevant_metadata = {k: patch_metadata[k] for k in ('name', 'namespace') if k in patch_metadata}
+
+        patched_resource = None
+
+        for yml_resource in k8s_resources:
+            resource_type, metadata = yml_resource['kind'], yml_resource['metadata']
+            relevant_metadata = {k: metadata[k] for k in ('name', 'namespace') if k in metadata}
+
+            if resource_type != patch_resource_type or relevant_metadata != patch_relevant_metadata:
+                continue
+            
+            patched_body = self.patch_resource_body(yml_resource, patch_body)
+            created_resource = self.create_from_dict(patched_body, **relevant_metadata)
+
+            done = self.wait_for_resource(resource_type=resource_type, status_list=["RUNNING", "ADDED"], **relevant_metadata)
+            if done:
+                patched_resource = created_resource
+
+            break
```

Comment:
It's correct! At this point it should stop trying whether or not the patch was successful.

Reply:
If you have several k8s_resources with relevant data it will stop after first "successful" iteration and other resources will not be patched. Logically, without lines 220-221, you starting **for** loop and finishes after first iteration.